docs/synchronization.md

```diff
@@ -0,0 +1,220 @@
+If a document has not been saved yet with actual synchronization data, we will have no way to retrieve it, except by providing a `findLegacyDocument` function to `saveEntry` options. This signature of a `findLegacyDocument` function is:
+```js
+async function (entry, cozy) => { /* look for your document */ }
+```
+The `cozy` parameter is an instance of Cozy-Client.
+
+The idea is for example to retrive a document by its file path or any other accurate information.
+
+### getSyncData
+This method returns any addtional synchronization data the konnector should need. Default method should be something returning an empty object or `null`.
+```js
+(entry) => ({})
+```
+A konnector should provide its own method:
+```js
+{
+  getSyncData: (entry) => ({
+    filename: entry.name
+  })
+}
+```
+
+### idAttribute
+#### String
+Name of the id attribute in entry. Default is `id`. If we want to use another name, we may use:
+```js
+saveFiles(files, {idAttribute: 'uuid'})
+```
+#### Array
+This attribute may also be an array of string, defining each entry attribute which must be used for identifier.
+
+##### Example
+```js
+saveFiles(files, {idAttribute: ['name', 'author', 'date']})
+```
+
+In this case, all fields are used in synchronization data.
+
+
+#### function
+
+When an entry does not provide an unique identifer, it is possible to use a function as `idAttribute` to customize the way the unique identifer is generated.
+
+##### Example
+```js
+saveFiles(files, {idAttribute: (entry) => hash(entry)})
+// hash() is a custom method  hashing the current file.
+```
+
+### konnector
+
+The konnector attribute is used as a key reference to retrieve all document synchronized by this konnector.
+
+For now, the `konnector` option will be needed for every call. But it could be interesting to initialize some kind of runner with it, which may encapsulate all the `cozy-konnector-libs` functions.
+
+### shouldSave
+
+The shouldSave function returns a boolean which indicate if the entry should be saved. The default method should be:
+```js
+async function shouldSave (entry, existingDocument, cozy) => {
+  return Promise.resolve(!existingDocument)
+}
+```
+When calling this function, it is assumed that `cozy-konnector-libs` first queries the database for an exisring document, based on synchronization data and pass the document resulting the query to the `shouldSave` function, event if this document is `null`.
+
+As a third parameter, a cozyClient instance is passed, it may be used by a konnector to perform another query.
+
+Because of this third parameter, the whole `shouldSave` method is asynchronous.
+
+### shouldUpdate
+
+The `shouldUpdate` function acts exactly like `shouldSave`, except that its used to determine if a document should be updated. By default, we make the choice to never update a document (except if it does not have synchronization data, see `findLegacyDocument` above). But some konnectors like `Linxo` related ones need to update existing documents.
```

Comment:
But the linxo connector does not save files.

Reply:
Well spotted. I don't know how bank operations are saved, could you link me the part in konnector-libs or in the Linxo konnector please ?

Reply:
I checked how Linxo works, and it is not incompatible, as a document here may have any doctype.

By the way, I am adding that we have to handle synchronisation options not only in `saveFiles` method but also in `addData`.